File: Tools/Cache Editor/src/alex/decompressors/GZipDecompressor.java
package alex.decompressors;

import java.util.zip.Inflater;

import alex.io.Stream;

public class GZipDecompressor {

	private static final Inflater inflater = new Inflater(true);

	public static final void decompress(Stream stream, byte output[]) {
		if (~stream.payload[stream.offset] != -32
				|| stream.payload[stream.offset + 1] != -117) {
			throw new RuntimeException("Invalid GZIP header!");
		}
		try {
			inflater.setInput(stream.payload, stream.offset + 10,
					-stream.offset - 18 + stream.payload.length);
			inflater.inflate(output);
		} catch (Exception _ex) {
			inflater.reset();
			throw new RuntimeException("Invalid GZIP compressed data!");
		}
		inflater.reset();
	}
}

File: Client/src/main/kotlin/org/rs09/client/filestore/compression/GzipDecompressor.kt
package org.rs09.client.filestore.compression

import org.rs09.client.DataBuffer
import java.util.zip.Inflater

object GzipDecompressor {
    private var inflater = Inflater(true)

    fun decompress(buffer: DataBuffer, out: ByteArray) {
        if (buffer.buffer[buffer.index].toInt() != 31 || buffer.buffer[buffer.index + 1].toInt() != -117)
            throw RuntimeException("Invalid GZIP header!")

        try {
            inflater.setInput(buffer.buffer, buffer.index + 10, -8 - (10 + buffer.index) + buffer.buffer.size)
            inflater.inflate(out)
        } catch (var5: Exception) {
            inflater.reset()
            throw RuntimeException("Invalid GZIP compressed data!")
        }
        inflater.reset()
    }
}